### PR TITLE
chore: Add support for custom network images

### PR DESCRIPTION
**Pull request by gambinish**
## **Description**

Resurrecting an older PR of mine to provide support for custom network images: https://github.com/MetaMask/metamask-mobile/pull/10448

## **Related issues**

Fixes: Support custom network icons on mobile.

## **Manual testing steps**

1. Add custom network (In this case Flare Mainnet or Songbird Testnet)
2. Icon should appear wherever Network Icon should appear

## **Screenshots/Recordings**

https://github.com/user-attachments/assets/6ea5b310-4b6a-4b6c-9656-d892bae83fc3

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/components/UI/Notification/NotificationMenuItem/__snapshots__/Content.test.tsx.snap [hunk 1]
```diff
@@ -47,7 +47,7 @@ exports[`NotificationContent renders correctly 1`] = `
         }
       }
     >
-      5 months ago
```
Comment (gambinish) [inline app/components/UI/Notification/NotificationMenuItem/__snapshots__/Content.test.tsx.snap:50]: No idea why this snapshot is updating here.

Comment (gambinish) [reply app/components/UI/Notification/NotificationMenuItem/__snapshots__/Content.test.tsx.snap:50]: Okay, I think this snapshot actually will update [once per month](https://github.com/MetaMask/metamask-mobile/blob/922ffbbd6fd53d104a42073a84fd6cdeca8b5e1e/app/components/UI/Notification/NotificationMenuItem/Content.test.tsx#L8)? Seems odd?

Comment (NicolasMassart) [reply app/components/UI/Notification/NotificationMenuItem/__snapshots__/Content.test.tsx.snap:50]: Wow that's a very bad pattern then!
We must make the date in the test generated to always provide the same snapshot. Using now() would be probably fine.
Of course it's out of the scope of your PR, so can you create an issue for that please?

File: app/util/networks/customNetworks.tsx [hunk 2]
```diff
@@ -129,3 +129,8 @@ export const UnpopularNetworkList = [
     },
   },
 ];
+
+export const CustomNetworkImgMapping = {
+  [toHex('14')]: require('../../images/flare-mainnet.png'), // Flare Mainnet
```
Comment (NicolasMassart) [inline app/util/networks/customNetworks.tsx:134]: Q: why not directly using hex values as keys?

File: app/components/UI/Notification/NotificationMenuItem/__snapshots__/Content.test.tsx.snap (hunk 1 above)
Comment (NicolasMassart) [reply app/components/UI/Notification/NotificationMenuItem/__snapshots__/Content.test.tsx.snap:50]: Created this issue https://github.com/MetaMask/metamask-mobile/issues/11785 as it's blocking everyone as unit tests all fail

File: app/util/networks/customNetworks.tsx (hunk 2 above)
Comment (gambinish) [reply app/util/networks/customNetworks.tsx:134]: We can use the hex values here. I'll make this change.

**Review by sahar-fehri — APPROVED** [review]

**Review by tommasini — APPROVED** [review]

**Review by salimtb — APPROVED** [review]

**Pull request merged**